aiohttp/connector.py

```diff
@@ -464,11 +487,18 @@ def _create_connection(self, req):
 
         for hinfo in hosts:
             try:
-                return (yield from self._loop.create_connection(
+                conn = yield from self._loop.create_connection(
                     self._factory, hinfo['host'], hinfo['port'],
                     ssl=sslcontext, family=hinfo['family'],
                     proto=hinfo['proto'], flags=hinfo['flags'],
-                    server_hostname=hinfo['hostname'] if sslcontext else None))
+                    server_hostname=hinfo['hostname'] if sslcontext else None)
+                if self._verify_fingerprint:
+                    sock = conn[0]._sock
+                    cert = sock.getpeercert(True)
+                    digest = self._hashfunc(cert).digest()
+                    if digest != self._fingerprint_bytes:
+                        raise FingerprintMismatch
```

Comment:
Use at least `FingerprintMismatch()`. Maybe later you'll add info about failing `host:port` pair -- it is helpful for understanding the source of problem.